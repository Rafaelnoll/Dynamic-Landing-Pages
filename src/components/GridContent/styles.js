import Styled, { css } from "styled-components";

export const Container = Styled.div`
    ${({ theme }) => css`
      text-align: center;
      max-width: 58rem;
      margin: 0 auto;
    `}
`;

export const Html = Styled.div`
    ${({ theme }) => css`
      margin: ${theme.spacings.xlarge} 0; 
    `}
`;
