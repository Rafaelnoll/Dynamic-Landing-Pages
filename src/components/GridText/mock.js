// eslint-disable-next-line import/no-anonymous-default-export
export default {
    background: false,
    title: 'My grid',
    description: 'Atque doloribus nobis l√ßasudatium esse ui, ut non commidu maxime distinctio veritats unde',
    grid:[
      {
         title: 'Teste 1',
         description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et metus arcu. Integer pulvinar hendrerit fermentum. Vestibulum faucibus dapibus erat, non porttitor ipsum vehicula a. Nam non ex dignissim, dictum justo eget, mattis ante. Donec aliquam vel turpis quis semper. Proin tincidunt semper metus et sagittis. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Proin fringilla finibus lacinia. Etiam non imperdiet mi, efficitur scelerisque arcu. Sed vel lectus nulla. Maecenas odio enim, viverra in viverra eget, luctus eget ipsum. Pellentesque lacinia, ante quis eleifend porttitor, purus dui auctor nisi, sed congue tellus tortor nec magna.'
      },
      {
         title: 'Teste 2',
         description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et metus arcu. Integer pulvinar hendrerit fermentum. Vestibulum faucibus dapibus erat, non porttitor ipsum vehicula a. Nam non ex dignissim, dictum justo eget, mattis ante. Donec aliquam vel turpis quis semper. Proin tincidunt semper metus et sagittis. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Proin fringilla finibus lacinia. Etiam non imperdiet mi, efficitur scelerisque arcu. Sed vel lectus nulla. Maecenas odio enim, viverra in viverra eget, luctus eget ipsum. Pellentesque lacinia, ante quis eleifend porttitor, purus dui auctor nisi, sed congue tellus tortor nec magna.'
      },
      {
         title: 'Teste 3',
         description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et metus arcu. Integer pulvinar hendrerit fermentum. Vestibulum faucibus dapibus erat, non porttitor ipsum vehicula a. Nam non ex dignissim, dictum justo eget, mattis ante. Donec aliquam vel turpis quis semper. Proin tincidunt semper metus et sagittis. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Proin fringilla finibus lacinia. Etiam non imperdiet mi, efficitur scelerisque arcu. Sed vel lectus nulla. Maecenas odio enim, viverra in viverra eget, luctus eget ipsum. Pellentesque lacinia, ante quis eleifend porttitor, purus dui auctor nisi, sed congue tellus tortor nec magna.'
      },
    ]
}