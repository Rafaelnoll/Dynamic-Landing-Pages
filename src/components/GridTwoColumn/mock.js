import JavascriptIcon from '../../assets/images/javascript.svg';

// eslint-disable-next-line import/no-anonymous-default-export
export default {
    title: "Grid Two Column",
    text: `Lorem ipsum dolor sit, amet consectetur adipisicing elit.
        Nemo ut culpa, ab architecto molestiae labore. Id molestiae nesciunt impedit, 
        veritatis ab vel natus necessitatibus, voluptatibus illum quas, architecto 
        debitis esse.`,
    srcImg: JavascriptIcon,
    background: false,
}