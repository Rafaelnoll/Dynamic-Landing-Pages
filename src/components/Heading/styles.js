import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.big};
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${titleMedia(theme)};
  `,
}

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? "uppercase" : "none"};
`;

const titleMedia = (theme) => css`
  @media ${theme.media.ltmedium} {
    font-size: ${theme.font.sizes.large};
  }
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;