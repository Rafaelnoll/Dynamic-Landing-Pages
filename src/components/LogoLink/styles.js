import styled, { css } from "styled-components";

export const Container = styled.a`
    ${({ theme }) => css`
        display: flex;
        justify-content: center;
        align-items: center;
        text-decoration: none;
        color: inherit;

        > img {
            max-height: 3rem;
        }
    `}
`