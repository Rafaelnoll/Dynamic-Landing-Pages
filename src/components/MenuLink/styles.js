import Styled, { css } from "styled-components";

export const Link = Styled.a`
    ${({ theme }) => css`
        display: block;
        font-size: ${theme.font.sizes.small};
        text-decoration: none;
        font-size: ${theme.font.sizes.small};
        padding: ${theme.spacings.small};
        color: ${theme.colors.primaryColor};
        position: relative;

        &::after{
            content: '';
            position: absolute;
            bottom: 0;
            left: 50%;
            width: 0;
            height: 0.2rem;
            background: ${theme.colors.secondaryColor};
            transition: all 300ms ease-in-out;
        }
        
        &:hover::after{
            left: 25%;
            width: 50%;
        }
    `};
`;