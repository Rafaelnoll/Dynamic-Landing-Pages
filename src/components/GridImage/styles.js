import Styled, { css } from "styled-components";
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = Styled.div`
    ${({ theme }) => css`
    > ${TextComponent}{
        margin-bottom: ${theme.spacings.xhuge};
      }
    `}
`;

export const Grid = Styled.div`
    ${({ theme }) => css`
      display: grid;
      grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
      gap: ${theme.spacings.large};

      @media ${theme.media.ltmedium} {
        grid-template-columns: 1fr;
      }
    `}
`;

export const GridElement = Styled.div`
    ${({ theme }) => css`
      overflow: hidden;
    `}
`;

export const Image = Styled.img`
    ${({ theme }) => css`
      width: 100%;
      transition: all 300ms ease-in-out;

      &:hover{
        transform: scale(1.2) rotate(10deg);
      }
    `}
`;

