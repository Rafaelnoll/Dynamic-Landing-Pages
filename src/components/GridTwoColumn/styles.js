import Styled, { css } from "styled-components";
import { Title } from '../Heading/styles';

export const Container = Styled.div`
    ${({ theme }) => css`
      display: grid;
      grid-template-columns: 1fr 2fr;
      align-items: center;
      gap: ${theme.spacings.large};

      @media ${theme.media.ltmedium}{
        grid-template-columns: 1fr;
        text-align: center;
      }

      ${Title}{
        margin-bottom: ${theme.spacings.large};
      }
    `}
`;

export const TextContainer = Styled.div`
    ${({ theme }) => css`
        @media ${theme.media.ltmedium}{
            margin-bottom: ${theme.spacings.large};
        }
    `}
`;

export const ImageContainer = Styled.div`
    ${({ theme }) => css`
      
    `}
`;

export const Image = Styled.img`
    ${({ theme }) => css`
      
    `}
`;