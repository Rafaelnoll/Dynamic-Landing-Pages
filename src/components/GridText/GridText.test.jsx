import { renderTheme } from "../../styles/render-theme";
import { GridText } from ".";
import mock from './mock';

describe("<GridText/>", () => {
    it("should render grid text", () => {
        const { container } = renderTheme(<GridText {...mock} background={undefined} />);
        expect(container.firstChild).toMatchSnapshot();
    });
})