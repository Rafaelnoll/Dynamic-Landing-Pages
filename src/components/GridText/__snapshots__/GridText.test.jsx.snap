// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridText/> should render grid text 1`] = `
.c4 {
  color: #0a1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c9 {
  color: #0a1128;
  font-size: 3.2rem;
  text-transform: none;
}

.c6 {
  font-size: 2.4rem;
}

.c2 > .c5 {
  margin-bottom: 6.4rem;
}

.c7 {
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 4.0rem;
}

.c8 .c3 {
  position: relative;
  left: 5rem;
}

.c8 .c3::before {
  counter-increment: grid-counter;
  content: counter(grid-counter);
  position: absolute;
  font-size: 7rem;
  top: -3rem;
  left: -5rem;
  -webkit-transform: rotate(10deg);
  -ms-transform: rotate(10deg);
  transform: rotate(10deg);
}

.c0 {
  background-color: #FFF;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 4.0rem;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4.0rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h2
        class="c3 c4"
      >
        My grid
      </h2>
      <div
        class="c5 c6"
      >
        Atque doloribus nobis l√ßasudatium esse ui, ut non commidu maxime distinctio veritats unde
      </div>
      <div
        class="c7"
      >
        <div
          class="c8"
        >
          <h3
            class="c3 c9"
          >
            Teste 1
          </h3>
          <div
            class="c5 c6"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et metus arcu. Integer pulvinar hendrerit fermentum. Vestibulum faucibus dapibus erat, non porttitor ipsum vehicula a. Nam non ex dignissim, dictum justo eget, mattis ante. Donec aliquam vel turpis quis semper. Proin tincidunt semper metus et sagittis. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Proin fringilla finibus lacinia. Etiam non imperdiet mi, efficitur scelerisque arcu. Sed vel lectus nulla. Maecenas odio enim, viverra in viverra eget, luctus eget ipsum. Pellentesque lacinia, ante quis eleifend porttitor, purus dui auctor nisi, sed congue tellus tortor nec magna.
          </div>
        </div>
        <div
          class="c8"
        >
          <h3
            class="c3 c9"
          >
            Teste 2
          </h3>
          <div
            class="c5 c6"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et metus arcu. Integer pulvinar hendrerit fermentum. Vestibulum faucibus dapibus erat, non porttitor ipsum vehicula a. Nam non ex dignissim, dictum justo eget, mattis ante. Donec aliquam vel turpis quis semper. Proin tincidunt semper metus et sagittis. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Proin fringilla finibus lacinia. Etiam non imperdiet mi, efficitur scelerisque arcu. Sed vel lectus nulla. Maecenas odio enim, viverra in viverra eget, luctus eget ipsum. Pellentesque lacinia, ante quis eleifend porttitor, purus dui auctor nisi, sed congue tellus tortor nec magna.
          </div>
        </div>
        <div
          class="c8"
        >
          <h3
            class="c3 c9"
          >
            Teste 3
          </h3>
          <div
            class="c5 c6"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et metus arcu. Integer pulvinar hendrerit fermentum. Vestibulum faucibus dapibus erat, non porttitor ipsum vehicula a. Nam non ex dignissim, dictum justo eget, mattis ante. Donec aliquam vel turpis quis semper. Proin tincidunt semper metus et sagittis. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Proin fringilla finibus lacinia. Etiam non imperdiet mi, efficitur scelerisque arcu. Sed vel lectus nulla. Maecenas odio enim, viverra in viverra eget, luctus eget ipsum. Pellentesque lacinia, ante quis eleifend porttitor, purus dui auctor nisi, sed congue tellus tortor nec magna.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
